### PR TITLE
79 min amount of images

### PR DESCRIPTION
**Pull request by thomaskettunen**
I made a bit of a mistake, the pull request is up again

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,7 @@
 /dataSources/
 /dataSources.local.xml
 
-**/image-*
+*/image_db
```
Comment (MartinCornelius) [inline .gitignore:10]: Skal være `**/image-*` så vi slipper for mappen.

File: .gitignore [hunk 2]
```diff
@@ -20,5 +20,3 @@ venv
 core/logs/*
 core/modelcheckpoint/*
 .conda/*
-
-core/latest.keras
```
Comment (MartinCornelius) [inline .gitignore:24]: Tilføj lige denne linje til PR ;)

File: core/data/fetch.py [hunk 3]
```diff
@@ -125,6 +130,58 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
         if extra:
             out += extra
         return out + "_0.npy"
+    @staticmethod
+    def pad_dataset(df, raw_dataset_path: str, raw_label_path: str, min_amount_of_pictures=3):
+        run_correction = False
+
+        print("coo coo", df)
```
Comment (MartinCornelius) [inline core/data/fetch.py:137]: Nej

File: core/data/fetch.py [hunk 4]
```diff
@@ -125,6 +130,58 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
...
+
+        itt = 0
+
+        less_than_list = []
+
+        for count in counts:
+            if min_amount_of_pictures > count:
+                less_than_list.append((values[itt], count))
+                run_correction = True
+            itt += 1
```
Comment (MartinCornelius) [inline core/data/fetch.py:150]: Evt. brug `for itt in range(len(counts))` i stedet for at have variabel optælling :)

File: core/data/fetch.py [hunk 5]
```diff
@@ -125,6 +130,58 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
...
+
+        if run_correction:
+            # just need to make it so that if there are species in the dataset, which are below the min_amount, then find them in the global dataset and put into the other one
```
Comment (MartinCornelius) [inline core/data/fetch.py:153]: Er kommentar fikset?

File: core/run.py [hunk 6]
```diff
@@ -19,9 +18,15 @@
                   ft_extractor=ft_extractor,
                   num_rows=num_rows,
                   degrees="none",
-                  bfly=["all"])
+                  bfly=["all"],
+                  min_amount_of_pictures=5)
```
Comment (MartinCornelius) [inline core/run.py:22]: Denne linje er relevant.

File: core/run.py [hunk 7]
```diff
@@ -19,9 +18,15 @@
...
     df = db.setup_dataset()
+    print("df_before: ", df)
```
Comment (MartinCornelius) [inline core/run.py:24]: Ryd lige op, tak!

File: core/util/ConvertWorldToCsv.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import pandas as pd
+
+
+# Converts the world data to have less useless rows
+def WorldConverter():
+    df = pd.DataFrame = pd.read_csv("leopidotera-world/occurrence.txt", sep="	", low_memory=False)
+    # df.to_csv("occurrence.csv", index=False)
+    print(df[df.columns])
+    df.drop(columns=[col for col in df if
```
Comment (MartinCornelius) [inline core/util/ConvertWorldToCsv.py:9]: Må gerne formateres

File: test/test_data/test_feature.py
Comment (MartinCornelius) [inline test/test_data/test_feature.py:1]: Pas på du ikke overrider med en gammel fil!

File: test/test_data/test_fetch.py
Comment (MartinCornelius) [inline test/test_data/test_fetch.py:1]: Pas på du ikke overrider med en gammel fil

**Review by MartinCornelius — DISMISSED** [review]

File: core/data/fetch.py [hunk 9]
```diff
@@ -11,12 +11,13 @@
...
 from core.util.logging.logable import Logable
 from core.util.util import timing, setup_log, log_ent_exit
-from core.util.constants import IMGDIR_PATH, MERGE_COLS, BFLY_FAMILY, BFLY_LIFESTAGE, DATASET_PATH, DIRNAME_DELIM
+from core.util.constants import IMGDIR_PATH, MERGE_COLS, BFLY_FAMILY, BFLY_LIFESTAGE, DATASET_PATH, DIRNAME_DELIM, RAW_WORLD_DATA_PATH, RAW_WORLD_LABEL_PATH
 from core.data.feature import FeatureExtractor
 
 
 class Database(Logable):
     def __init__(self,
+                 min_amount_of_pictures,
```
Comment (khmikkelsen) [inline core/data/fetch.py:20]: Didn't we agree to also include number of species? For experimental sake, we could see the effect on model precision, if we include more images. We dont have unlimited GRAM.

File: core/data/fetch.py [hunk 10]
```diff
@@ -125,6 +130,54 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
...
+            print("Total number of extra rows: ", totalRows)
+            world_df = world_df.loc[world_df['species'].isin(species_with_less_than_optimal_amount_of_images)]
+            # loop that gets the species, which are below the required amount
+            for item, count in less_than_list:
+                # print("World pictures to add: ", min_amount_of_pictures - count)
+                world_specific = world_df.loc[world_df["species"] == item]
+                world_specific = world_specific.iloc[:min_amount_of_pictures - count]
+                df = df.append(world_specific, ignore_index=True)
+            # df = Database.fetch_images(df, "identifier")
+        return df
```
Comment (khmikkelsen) [inline core/data/fetch.py:180]: Why are we checking whether min amount has been reached? We can merge world into dk dataset based on species, and done deal, and then drop rows when the min amount of each species as been met. We don't care if the picture has been taken in dk, just that the species has occurred in denmark.

File: core/run.py [hunk 11]
```diff
@@ -30,3 +31,4 @@
     model.fit(50)  # Epochs
     model.save()
     # model.evaluate_and_show_predictions(num_samples=3)
+    '''
```
Comment (khmikkelsen) [inline core/run.py:34]: Clean up please. Main should call fit.

File: core/util/ConvertWorldToCsv.py [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import pandas as pd
+
+
+# Converts the world data to have less useless rows
+def WorldConverter():
```
Comment (khmikkelsen) [inline core/util/ConvertWorldToCsv.py:5]: ayayay caramba, that is not snake case compliant 🤢🤢

**Review by khmikkelsen — CHANGES_REQUESTED** [review]

File: core/data/fetch.py [hunk 13]
```diff
@@ -125,6 +130,54 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
         if extra:
             out += extra
         return out + "_0.npy"
+    @staticmethod
```
Comment (MartinCornelius) [inline core/data/fetch.py:133]: Formattering

File: core/data/fetch.py [hunk 14]
```diff
@@ -125,6 +130,54 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
...
+
+            totalRows = totalRows * min_amount_of_pictures
+            print("Total number of extra rows: ", totalRows)
+            world_df = world_df.loc[world_df['species'].isin(species_with_less_than_optimal_amount_of_images)]
+            # loop that gets the species, which are below the required amount
+            for item, count in less_than_list:
+                # print("World pictures to add: ", min_amount_of_pictures - count)
+                world_specific = world_df.loc[world_df["species"] == item]
+                world_specific = world_specific.iloc[:min_amount_of_pictures - count]
+                df = df.append(world_specific, ignore_index=True)
```
Comment (MartinCornelius) [inline core/data/fetch.py:178]: Er det meningen, at der skal bruges `df.append` her?

Comment (khmikkelsen) [reply core/data/fetch.py:178]: dataframe.append is deprecated in numpy 1.24.3

**Comment by khmikkelsen** [comment]
Please mark the fixed change requests as resolved when they're done.

**Pull request merged**